distribution/tools/entitlements-agent/build.gradle

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+apply plugin: 'elasticsearch.build'
+
+configurations {
+  entitlementRuntime
+}
+
+dependencies {
+  entitlementRuntime project(":libs:elasticsearch-entitlements-runtime")
+  implementation project(":libs:elasticsearch-entitlements-runtime")
+  testImplementation project(":test:framework")
+}
+
+test {
+  dependsOn(jar)
+  jvmArgs "-javaagent:${jar.archiveFile.get().asFile}"
```

Comment:
I don't think this will work. The string interpolation will happen at gradle configuration time, but it needs to be delayed to runtime. In the past I've done hacks with GString like `"${-> '-javaagent:' + jar.archiveFile.get().asFile}`, but I'm sure @mark-vieira has a better way to do that lazy evaluation now.

Reply:
Is this supposed to start ES with the agent? If so, how we would do that? 
I think it's a good idea to add a minimal README.md (just a few lines atm with the basic purpose of this entitlements-agent project and how to launch it).

Reply:
Does it need to be delayed to runtime if it's only for the agent's own unit tests?

Reply:
@rjernst - can I merge it like this? Or does it need to be changed?

Reply:
@breskeby - is this something you know about?

Reply:
you applied my suggestion and I verified the fix locally seeing the agent is correctly applied to the tests and we didn't break task avoidance: https://gradle-enterprise.elastic.co/s/q7l56qcrj2cbm/performance/configuration/by-project?openProjects=WyJwcm9qZWN0LTpkaXN0cmlidXRpb246dG9vbHM6ZW50aXRsZW1lbnQtYWdlbnQiXQ#project-:distribution:tools:entitlement-agent 
👍

Reply:
> can I merge it like this? Or does it need to be changed?

Two jars seem like an unnecessary distinction to me. But I'm ok with merging as-is and debating that separately.